src/extension/geofence-server/src/test/java/org/geoserver/geofence/integration/GeofenceAccessManagerIntegrationTest.java

```diff
@@ -708,6 +729,313 @@ public void testAdminRule_WorkspaceAccessLimits_Username_based_rule() {
+        LayerGroupInfo group1 = null;
+        LayerGroupInfo group2 = null;
+        try {
+            Authentication user = getUser("anonymousUser", "", "ROLE_ANONYMOUS");
+            login("admin", "geoserver", new String[] {"ROLE_ADMINISTRATOR"});
+            Catalog catalog = getCatalog();
+
+            LayerInfo places = catalog.getLayerByName(getLayerId(MockData.NAMED_PLACES));
+            LayerInfo forests = catalog.getLayerByName(getLayerId(MockData.FORESTS));
+
+            group1 =
+                    createsLayerGroup(
+                            catalog,
+                            "group21",
+                            LayerGroupInfo.Mode.SINGLE,
+                            null,
+                            Arrays.asList(places, forests));
+            group2 =
+                    createsLayerGroup(
+                            catalog,
+                            "group22",
+                            LayerGroupInfo.Mode.NAMED,
+                            null,
+                            Arrays.asList(places, forests));
+            // limit rule for anonymousUser on LayerGroup group1
+            idRule =
+                    support.addRule(
+                            GrantType.LIMIT,
+                            "anonymousUser",
+                            "ROLE_ANONYMOUS",
+                            "WMS",
+                            null,
+                            null,
+                            "group21",
+                            0);
+
+            // limit rule for anonymousUser on LayerGroup group2
+            support.addRule(
+                    GrantType.LIMIT,
+                    "anonymousUser",
+                    "ROLE_ANONYMOUS",
+                    "WMS",
+                    null,
+                    null,
+                    "group22",
+                    1);
+
+            // add allowed Area only to the first layer group
+            support.addRuleLimits(idRule, CatalogMode.HIDE, AREA_WKT, 4326);
+
+            // mock a WMS request to check contained layers direct access
+            Request req = new Request();
+            req.setService("WMS");
+            req.setRequest("GetMap");
+            Dispatcher.REQUEST.set(req);
+            logout();
+
+            login("anonymousUser", "", new String[] {"ROLE_ANONYMOUS"});
+
+            VectorAccessLimits vl =
+                    (VectorAccessLimits) accessManager.getAccessLimits(user, places);
+
+            assertEquals(vl.getReadFilter(), Filter.INCLUDE);
+            assertEquals(vl.getWriteFilter(), Filter.INCLUDE);
+            logout();
+        } finally {
+            removeLayerGroup(group1, group2);
+        }
+    }
+
+    @Test
+    public void testLayerInGroupAreaEnlargement() throws Exception {
+        // tests that when a Layer is directly accessed for WMS request
+        // if it is belonging to a group and both the group and the layer have rule with allowed
+        // area,
```

Comment:
Comment can be rewrapped. In general, try to collaborate with the formatter...

Reply:
fixed